### ISSUE TITLE
[BUG] ETXTBSY with README example

### ISSUE THREAD

**Issue by yujiri8** [alixaxel/chrome-aws-lambda#209]
## Environment
* `chrome-aws-lambda` Version: 8.0.1
* `puppeteer` / `puppeteer-core` Version: 8.0.0
* OS: Linux
* Node.js Version: 14.x
* Lambda / GCF Runtime: `Node.js 14.x`

## Expected Behavior

Chromium launches successfully.

## Current Behavior

The following error returned when testing in AWS console:
```
{
  "errorType": "Error",
  "errorMessage": "spawn ETXTBSY",
  "trace": [
    "Error: spawn ETXTBSY",
    "    at ChildProcess.spawn (internal/child_process.js:403:11)",
    "    at Object.spawn (child_process.js:553:9)",
    "    at BrowserRunner.start (/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserRunner.js:79:34)",
    "    at ChromeLauncher.launch (/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/Launcher.js:86:16)",
    "    at async Runtime.exports.handler (/var/task/index.js:8:15)"
  ]
}```

## Steps to Reproduce

I copied exactly the example from the README:

```js
const chromium = require('chrome-aws-lambda');

exports.handler = async (event, context, callback) => {
  let result = null;
  let browser = null;

  try {
    browser = await chromium.puppeteer.launch({
      args: chromium.args,
      defaultViewport: chromium.defaultViewport,
      executablePath: await chromium.executablePath,
      headless: chromium.headless,
      ignoreHTTPSErrors: true,
    });

    let page = await browser.newPage();

    await page.goto(event.url || 'https://example.com');

    result = await page.title();
  } catch (error) {
    return callback(error);
  } finally {
    if (browser !== null) {
      await browser.close();
    }
  }

  return callback(null, result);
};
```

I tried the answer [here](https://stackoverflow.com/a/62362152/12211329), clearing my node_modules and running `npm install --no-bin-links`, and it had no effect for me.

**Comment by alixaxel** [comment]
@yujiri8 That's very odd - the error code `ETXTBSY` means that something appears to be trying to write over the Chromium binary on `/tmp/chromium` while puppeteer is already executing it. The binary is inflated by calling `chromium.executablePath`, but you are correctly waiting for this promise to resolve and it doesn't seem like you're calling this getter elsewhere. 🤔 

If you invoke the Lambda again, do you get the same error repeatedly?

**Comment by yujiri8** [comment]
> If you invoke the Lambda again, do you get the same error repeatedly?

Yes.

**Comment by alixaxel** [comment]
@yujiri8 I'm unable to reproduce the problem. Are you able to provide me with a ZIP of your Lambda?

**Comment by yujiri8** [comment]
The error is different today. Now when I upload the same code, it gives a timeout on first run and a "failed to launch browser" on second run:
```
{
  "errorType": "Error",
  "errorMessage": "Failed to launch the browser process!\n\n\nTROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md\n",
  "trace": [
    "Error: Failed to launch the browser process!",
    "",
    "",
    "TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md",
    "",
    "    at onClose (/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserRunner.js:193:20)",
    "    at Interface.<anonymous> (/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserRunner.js:183:68)",
    "    at Interface.emit (events.js:327:22)",
    "    at Interface.close (readline.js:424:8)",
    "    at Socket.onend (readline.js:202:10)",
    "    at Socket.emit (events.js:327:22)",
    "    at endReadableNT (internal/streams/readable.js:1327:12)",
    "    at processTicksAndRejections (internal/process/task_queues.js:80:21)"
  ]
}
```
It may be because I hadn't kept the original zip around and recreated it this morning, and something was updated. I did notice the latest chrome-aws-lambda version was 8.0.2 instead of 8.0.1 which I think it was last night, but even grabbing the older version gives the new error now.

[Here](https://yujiri.xyz/code.zip) is a zip of the lambda I'm using now.

**Comment by alixaxel** [comment]
@yujiri8 I uploaded your Zip file and it ran successfully for me:

![image](https://user-images.githubusercontent.com/262782/113481159-dafa4100-948f-11eb-9d15-3f13435eec29.png)

Are you allocating enough memory (> 512MB) to your Lambda?

---

As for 8.0.2, it only introduced a fix to the Makefile and an additional overloaded method, so it can't be the culprit of this.

**Comment by yujiri8** [comment]
TL;DR my problem is solved.

I hadn't thought of the memory limit. Changing it from 128MB to 1024MB led to this:
```
START RequestId: 1094543c-0295-46d3-98ff-6df94c41dd8b Version: $LATEST
END RequestId: 1094543c-0295-46d3-98ff-6df94c41dd8b
REPORT RequestId: 1094543c-0295-46d3-98ff-6df94c41dd8b	Duration: 3003.58 ms	Billed Duration: 3000 ms	Memory Size: 1024 MB	Max Memory Used: 398 MB	Init Duration: 184.45 ms	
2021-04-03T15:37:52.790Z 1094543c-0295-46d3-98ff-6df94c41dd8b Task timed out after 3.00 seconds
```
And on second run:
```

**Comment by yujiri8** [comment]
START RequestId: 78c3e307-17d9-461b-b338-66ea30d48ba2 Version: $LATEST
2021-04-03T15:38:13.624Z	78c3e307-17d9-461b-b338-66ea30d48ba2	ERROR	Invoke Error 	{"errorType":"Error","errorMessage":"Failed to launch the browser process!\n\n\nTROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md\n","stack":["Error: Failed to launch the browser process!","","","TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md","","    at onClose (/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserRunner.js:193:20)","    at Interface.<anonymous> (/var/task/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserRunner.js:183:68)","    at Interface.emit (events.js:327:22)","    at Interface.close (readline.js:424:8)","    at Socket.onend (readline.js:202:10)","    at Socket.emit (events.js:327:22)","    at endReadableNT (internal/streams/readable.js:1327:12)","    at processTicksAndRejections (internal/process/task_queues.js:80:21)"]}
END RequestId: 78c3e307-17d9-461b-b338-66ea30d48ba2
REPORT RequestId: 78c3e307-17d9-461b-b338-66ea30d48ba2	Duration: 298.83 ms	Billed Duration: 299 ms	Memory Size: 1024 MB	Max Memory Used: 28 MB	
```
But if I also increase the timeout, it succeeds in:
```
START RequestId: fb3ce2bb-3708-42a5-88a4-fe8cdd9d908d Version: $LATEST
END RequestId: fb3ce2bb-3708-42a5-88a4-fe8cdd9d908d
REPORT RequestId: fb3ce2bb-3708-42a5-88a4-fe8cdd9d908d	Duration: 3887.04 ms	Billed Duration: 3888 ms	Memory Size: 1024 MB	Max Memory Used: 437 MB	Init Duration: 216.92 ms
```

I'll close the issue. Thanks for your help!
